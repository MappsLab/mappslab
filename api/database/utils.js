// @flow

import * as R from 'ramda'
import type { Filter, PaginationArgs } from 'Types/sharedTypes'

export const createVariables = R.pipe(
	R.toPairs,
	R.reduce(
		(acc, [key, value]) =>
			// if the value is undefined or null,
			value === undefined || value === null
				? // return the accumulator
				  acc
				: // Else, return the value cast to a string
				  R.assoc(`$${key}`, value.toString())(acc),
		{},
	),
)

export const defaultPaginationArgs = {
	first: 50,
	after: '0x0',
}

export const makePaginationArgs = (args: PaginationArgs): PaginationArgs => {
	// Adds +1 to the 'first' argument, we need this to see if there is a next page.
	const { first, ...rest } = { ...defaultPaginationArgs, ...args }
	return {
		first: first + 1,
		...rest,
	}
}

export const createFilterString = (filter: Array<Filter>, connect: string = 'AND'): string => {
	const funcs = filter.map(({ key, value, operator = 'eq' }) => `${operator}(${key}, "${value}")`)
	return `@filter(${funcs.join(connect)})`
}

export const validateUid = (uid: string): boolean => /^0x[\d\w]+$/.test(uid)

export const validateUidOrWildcard = (uid: string): boolean => uid === '*' || validateUid(uid)
