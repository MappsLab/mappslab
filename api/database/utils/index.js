export { default as makeFilterString } from './makeFilterString'
export * from './dbUtils'
