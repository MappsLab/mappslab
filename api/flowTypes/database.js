// @flow
import dbClient from 'Database/client'

export type DBEdge = {
	fromUid: string,
	pred: string,
	toUid: string,
}

export type PartialEdge =
	| {
			fromUid: string,
			pred: string,
	  }
	| {
			toUid: string,
			pred: string,
	  }

export type EdgeConfig = {
	unique?: boolean,
}

export type Txn = dbClient.newTxn

export type DBNode = {
	uid: string,
}

export type TxnWithNode = {
	txn: Txn,
	data: DBNode,
}
