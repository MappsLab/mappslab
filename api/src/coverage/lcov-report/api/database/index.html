<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for api/database</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> api/database
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">70.83% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>119/168</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.58% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>43/76</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.3% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>107/148</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="client.js"><a href="client.js.html">client.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="createEdge.js"><a href="createEdge.js.html">createEdge.js</a></td>
	<td data-value="80.95" class="pic high"><div class="chart"><div class="cover-fill" style="width: 80%;"></div><div class="cover-empty" style="width:20%;"></div></div></td>
	<td data-value="80.95" class="pct high">80.95%</td>
	<td data-value="21" class="abs high">17/21</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="78.95" class="pct medium">78.95%</td>
	<td data-value="19" class="abs medium">15/19</td>
	</tr>

<tr>
	<td class="file low" data-value="createNode.js"><a href="createNode.js.html">createNode.js</a></td>
	<td data-value="17.65" class="pic low"><div class="chart"><div class="cover-fill" style="width: 17%;"></div><div class="cover-empty" style="width:83%;"></div></div></td>
	<td data-value="17.65" class="pct low">17.65%</td>
	<td data-value="17" class="abs low">3/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	</tr>

<tr>
	<td class="file high" data-value="createNodeWithEdges.js"><a href="createNodeWithEdges.js.html">createNodeWithEdges.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="27" class="abs high">24/27</td>
	<td data-value="64.29" class="pct medium">64.29%</td>
	<td data-value="14" class="abs medium">9/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="88" class="pct high">88%</td>
	<td data-value="25" class="abs high">22/25</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="63.64" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 63%;"></div><div class="cover-empty" style="width:37%;"></div></div></td>
	<td data-value="63.64" class="pct medium">63.64%</td>
	<td data-value="11" class="abs medium">7/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

<tr>
	<td class="file high" data-value="mutateNode.js"><a href="mutateNode.js.html">mutateNode.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="27" class="abs high">24/27</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="14" class="abs medium">10/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="91.3" class="pct high">91.3%</td>
	<td data-value="23" class="abs high">21/23</td>
	</tr>

<tr>
	<td class="file high" data-value="removeEdge.js"><a href="removeEdge.js.html">removeEdge.js</a></td>
	<td data-value="80.77" class="pic high"><div class="chart"><div class="cover-fill" style="width: 80%;"></div><div class="cover-empty" style="width:20%;"></div></div></td>
	<td data-value="80.77" class="pct high">80.77%</td>
	<td data-value="26" class="abs high">21/26</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="86.36" class="pct high">86.36%</td>
	<td data-value="22" class="abs high">19/22</td>
	</tr>

<tr>
	<td class="file high" data-value="removeNode.js"><a href="removeNode.js.html">removeNode.js</a></td>
	<td data-value="87.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="16" class="abs high">14/16</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file low" data-value="removeNodeWithEdges.js"><a href="removeNodeWithEdges.js.html">removeNodeWithEdges.js</a></td>
	<td data-value="12.5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="16" class="abs low">2/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="15.38" class="pct low">15.38%</td>
	<td data-value="13" class="abs low">2/13</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 20:15:34 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
