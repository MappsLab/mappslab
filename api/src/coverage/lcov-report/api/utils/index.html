<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for api/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> api/utils
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.21% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>48/57</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.85% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>28/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>18/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>40/47</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="auth.js"><a href="auth.js.html">auth.js</a></td>
	<td data-value="85" class="pic high"><div class="chart"><div class="cover-fill" style="width: 85%;"></div><div class="cover-empty" style="width:15%;"></div></div></td>
	<td data-value="85" class="pct high">85%</td>
	<td data-value="20" class="abs high">17/20</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="19" class="abs high">16/19</td>
	</tr>

<tr>
	<td class="file high" data-value="graphql.js"><a href="graphql.js.html">graphql.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="92.31" class="pct high">92.31%</td>
	<td data-value="13" class="abs high">12/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="81.25" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="16" class="abs high">13/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	</tr>

<tr>
	<td class="file medium" data-value="parsing.js"><a href="parsing.js.html">parsing.js</a></td>
	<td data-value="72.73" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 72%;"></div><div class="cover-empty" style="width:28%;"></div></div></td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="11" class="abs medium">8/11</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 20:15:34 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
