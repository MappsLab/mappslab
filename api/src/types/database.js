// @flow
import dbClient from 'Database/client'
import type { MapType } from 'Types/MapTypes'
import type { UserType } from 'Types/UserTypes'
import type { ClassroomType } from 'Types/ClassroomTypes'
import type { PinType } from 'Types/PinTypes'

export type Facet = { [key: string]: string | boolean | number | Date }

export type DBEdge = {
	fromUid: string,
	pred: string,
	toUid: string,
	facets?: Array<Facet>,
}

export type PartialEdge =
	| {
			fromUid: string,
			pred: string,
			facets?: Array<Facet>,
	  }
	| {
			toUid: string,
			pred: string,
			facets?: Array<Facet>,
	  }

export type EdgeConfig = {
	unique?: boolean,
}

export type EdgeInput = [PartialEdge, EdgeConfig]

export type Txn = dbClient.newTxn

export type DBNode = MapType | UserType | ClassroomType | PinType

export type TxnWithNode = {
	txn: Txn,
	data: DBNode,
}
