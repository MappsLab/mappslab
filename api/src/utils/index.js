// @flow
import { pipeP, flatten, filter, assoc, prop } from 'ramda'
import type { NestedArray } from 'ramda'

export const filterNullAndUndefined = (obj: Object): Promise<Object> =>
	new Promise((resolve) =>
		resolve(filter((a) => a !== null && a !== undefined)(obj)),
	)

// Makes regular functions async for promisePipe
export const promisify = (f: Function): Function => async (...args) =>
	f(...args)

export const promisePipe = (...funcs: Array<Function>) =>
	pipeP(...funcs.map(promisify))

export const arrayify = (...things: NestedArray<mixed>): Array<mixed> =>
	flatten(things)

export const slugify = (text: string) =>
	text
		.toString()
		.toLowerCase()
		.replace(/\s+/g, '-')
		.replace(/[^\w-]+/g, '')
		.replace(/--+/g, '-')
		.replace(/^-+/, '')
		.replace(/-+$/, '')

export const createSlugFrom = (key: string) => (obj: Object) =>
	assoc('slug', slugify(prop(key, obj)))(obj)
