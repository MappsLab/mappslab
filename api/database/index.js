// @flow
import { mapObjIndexed } from 'ramda'
import { unflatten } from 'flat'
import createEdge from './createEdge'
import createNode from './createNode'
import createNodeWithEdge from './createNodeWithEdge'
import mutateNode from './mutateNode'
import dbClient from './client'
import { createVariables } from './utils'

// const debug = require('debug')('api')

/* Shortcuts */

const query = async (queryString: string, vars?: Object): Promise<Object | null> => {
	const q = vars
		? // if we have vars, parse them
		  dbClient.newTxn().queryWithVars(queryString, createVariables(vars))
		: // Otherwise, normal query
		  dbClient.newTxn().query(queryString)
	const result = await q
	return result ? mapObjIndexed((value) => value.map(unflatten))(result.getJson()) : null
}

module.exports = {
	createEdge,
	createNode,
	createNodeWithEdge,
	mutateNode,
	dbClient,
	query,
}
