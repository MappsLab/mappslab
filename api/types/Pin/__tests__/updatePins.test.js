/* eslint-disable no-undef */
import { request, getViewerForContext } from '../../../__tests__/utils'
import { removeNode, removeEdge } from '../../../database'
import { getFirstPins } from './utils'

let firstPins

let viewer

beforeAll(async (done) => {
	firstPins = await getFirstPins()
	viewer = await getViewerForContext()

	done()
})

const query = /* GraphQL */ `
	query pin($uid: String!) {
		pin(input: { uid: $uid }) {
			uid
			title
			owner {
				uid
				name
			}
		}
	}
`

const mutation = /* GraphQL */ `
	mutation UpdatePin($uid: String!, $title: String, $lat: Float, $lang: Float, $mapUids: [String], $lessonUids: [String]) {
		updatePin(input: { uid: $uid, title: $title, lat: $lat, lang: $lang, mapUids: $mapUids, lessonUids: $lessonUids }) {
			uid
			title
			lat
			lang
			owner {
				uid
				name
			}
		}
	}
`

const getVariables = () => ({
	uid: firstPins[0].uid,
	title: 'great new title, thank you',
})

describe('[updatePin]', () => {
	it('should return an error if there is no current viewer', async () => {
		const variables = getVariables()

		const result = await request(mutation, { variables })
		expect(result.errors).toMatchSnapshot()
	})

	it('should return an error if the viewer is not the owner of the pin', async () => {
		const variables = getVariables()
		const context = { viewer }
		const result = await request(mutation, { variables, context })
		expect(result.errors).toMatchSnapshot()
	})

	it('should properly update a pin', async () => {
		// save the original pin so we can reset
		const originalPinResult = await request(query, { variables: { uid: firstPins[0].uid } })
		const originalPin = originalPinResult.data.pin

		// Attempt updating
		const variables = getVariables()
		const context = { viewer: firstPins[0].owner }
		const result = await request(mutation, { variables, context })
		expect(result.data.updatePin.title).toBe(variables.title)

		// Reset the pin
		const reset = await request(mutation, { variables: originalPin, context })
		expect(reset.data.updatePin.title).toBe(firstPins[0].title)
	})

	// it('should update a pin', async () => {
	// 	const originalPin =
	//  /* Arrange */
	//  // const { container, getByTestId } = render( ... )
	//  /* Act */

	//  /* Assert */
	//  // expect(...)
	// })
})
