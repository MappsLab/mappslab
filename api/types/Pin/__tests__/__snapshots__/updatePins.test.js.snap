// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[updatePin] should return an error if the viewer is not the owner of the pin 1`] = `
Array [
  [GraphQLError: You can only update your own pins],
]
`;

exports[`[updatePin] should return an error if there is no current viewer 1`] = `
Array [
  [GraphQLError: You must be logged in to create new pins. Please log in and try again.],
]
`;
